src/twisted/internet/base.py

```diff
@@ -506,12 +659,31 @@ def installWaker(self):
         raise NotImplementedError(
             reflect.qual(self.__class__) + " did not implement installWaker")
 
+
     def installResolver(self, resolver):
-        assert IResolverSimple.providedBy(resolver)
```

Comment:
This could probably use a comment explaining how this object is being type-checked now.